steward/src/cork/proposals.rs

```diff
@@ -261,44 +256,51 @@ async fn poll_approved_cork_proposals(state: &mut ProposalThreadState) -> Result
             proposal.proposal_id, encoded_call
         );
 
-        if let Err(err) = schedule_cork(&cellar_id, encoded_call.clone(), block_height).await {
-            match confirm_sheduling(state, &cellar_id, encoded_call, block_height).await {
-                Ok(confirmed) => {
-                    if confirmed {
-                        info!(
-                            "call for cellar {} scheduled for block {} by proposal {}",
-                            cellar_id, block_height, proposal.proposal_id
-                        );
-                        state.increment_proposal_id();
-                        continue;
+        // retry scheduling call on errors up to configured amount of retries
+        let mut attempts = 0;
+        loop {
+            if let Err(err) = schedule_cork(&cellar_id, encoded_call.clone(), block_height).await {
+                error!("error scheduling cork. confirming if the cork was scheduled in case this is a timeout bug: {}", err);
+                match confirm_sheduling(state, &cellar_id, encoded_call.clone(), block_height).await
+                {
+                    Ok(confirmed) => {
+                        if confirmed {
+                            info!(
+                                "call for cellar {} scheduled for block {} by proposal {}",
+                                cellar_id, block_height, proposal.proposal_id
+                            );
+                            state.increment_proposal_id();
+                            break;
+                        }
                     }
-                }
-                Err(err) => {
-                    return Err(proposal_processing_error(format!(
-                        "failed to confirm cork scheduling for proposal {} reason: {}",
-                        proposal.proposal_id, err
-                    )))
-                }
-            };
+                    Err(err) => {
+                        attempts += 1;
+                        if attempts > config.cork.max_scheduling_retries {
+                            return Err(proposal_processing_error(format!(
```

Comment:
I think we need to increment the ID here before returning after `max_scheduling_retries` has been reached.